dist/flx/lib/interpreter.dart

```diff
@@ -18,23 +18,31 @@ String _getUrl({
 }
 
 class CallInterpretation {
-  String get callName => _data['call_name'];
-  String get ruleName => _data['rule_name'];
-  String get knowledge => _data['knowledge_string'];
+  const CallInterpretation({
+    this.ruleName,
+    this.knowledge,
+    this.call
+  });
 
-  Call get call => new Call.fromName(callName);
-  bool get hasInterpretation => ruleName != null && knowledge != null;
-
-  CallInterpretation._(this._data);
+  final String ruleName;
+  final String knowledge;
+  final Call call;
 
-  final Map<String, String> _data;
+  bool get hasInterpretation => ruleName != null && knowledge != null;
 }
 
-Future<List<CallInterpretation>> getInterpretations(CallHistory callHistory) async {
-  String url = _getUrl(callString: callHistory.calls.join(','));
-  List interpretations = JSON.decode(await http.read(url));
+final Map<String, Future<List<CallInterpretation>>> _memoryCache =
+    new Map<String, Future<List<CallInterpretation>>>();
 
-  return interpretations.map((item) {
-    return new CallInterpretation._(item);
-  }).toList();
+Future<List<CallInterpretation>> getInterpretations(CallHistory callHistory) {
+  String url = _getUrl(callString: callHistory.calls.join(','));
+  return _memoryCache.putIfAbsent(url, () async {
```

Comment:
Flutter needs a nice caching primitive in general.  Presumably someone has written this for Dart.  You configure it with your sizes, etc, and then just put as needed and it knows how to purge, etc.